osmomath/math.go

```diff
@@ -203,3 +213,46 @@ func OrderOfMagnitude(num Dec) int {
 
 	return order
 }
+
+// GetPrecomputeOrderOfMagnitude returns the order of magnitude of the given amount.
+// Uses look up table for precomputed order of magnitudes.
+// CONTRACT: amount must be positive or zero. Panics if not
+func GetPrecomputeOrderOfMagnitude(amount Int) int {
+	if amount.IsNegative() {
+		panic(fmt.Errorf("amount must be positive or zero, was (%s)", amount))
+	}
+
+	if amount.GT(TenE9) {
+		a := amount.Quo(TenE9)
+		return 9 + GetPrecomputeOrderOfMagnitude(a)
+	}
+	if amount.GTE(TenE9) {
+		return 9
+	}
+	if amount.GTE(TenE8) {
+		return 8
+	}
+	if amount.GTE(TenE7) {
+		return 7
+	}
+	if amount.GTE(TenE6) {
+		return 6
+	}
+	if amount.GTE(TenE5) {
+		return 5
+	}
+	if amount.GTE(TenE4) {
+		return 4
+	}
+	if amount.GTE(TenE3) {
+		return 3
+	}
+	if amount.GTE(TenE2) {
+		return 2
+	}
+	if amount.GTE(TenE1) {
+		return 1
+	}
+
```

Comment:
If you want, you can do this based on bitlen and comp's on the bitlen in the future rather than 10 BigInt comparisons

Reply:
e.g. do one bitlen call, do a switch based on bitlen, then do one comparison to a bigint to get the precise bound